### PR TITLE
ss-262: add email sending service

### PR DESCRIPTION
**Pull request by Evcat6**

### CODE REVIEW

**Review by anton-otroshchenko — APPROVED** [review]

File: backend/src/bundles/auth/auth.service.ts [hunk 1]
```diff
@@ -37,14 +39,33 @@ class AuthService {
...
+            const { id, email } = newUser.toObject();
             const token = this.tokenService.createToken({ id });
+            void this.sendAfterSignUpEmail(email);
             return {
                 token,
             };
         }
     }
 
+    private async sendAfterSignUpEmail(email: string): Promise<void> {
```
Comment (VItaliy-Kalenichenko) [inline backend/src/bundles/auth/auth.service.ts:51]: Move it to a separate service

Comment (VItaliy-Kalenichenko) [inline backend/src/bundles/auth/auth.service.ts:51]: Move it to a separate service

File: backend/src/bundles/auth/auth.service.ts [hunk 2]
```diff
@@ -37,14 +39,33 @@ class AuthService {
...
+        const htmlToSend = getTemplate({
+            name: 'sign-up-email-template',
+            context: {
+                title: 'SmartSpend',
+                dashboardLink: 'https://smartspend.netlify.app/dashboard',
```
Comment (VItaliy-Kalenichenko) [inline backend/src/bundles/auth/auth.service.ts:56]: I think it's better to move `https://smartspend.netlify.app`  to .env

File: backend/src/views/sign-up-email-template.hbs
Comment (VItaliy-Kalenichenko) [inline backend/src/views/sign-up-email-template.hbs:1]: Maybe views is not the best name for the folder with email templates

Comment (Evcat6) [reply backend/src/views/sign-up-email-template.hbs:1]: Resolved in [fa46b57](https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/283/commits/fa46b57150fc43acee5f216bad37172627fdb6b8)

File: backend/src/bundles/auth/auth.service.ts (hunk 2 above)
Comment (Evcat6) [reply backend/src/bundles/auth/auth.service.ts]: Resolved in [89e9e0b](https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/283/commits/89e9e0b34988b69c1d7370414a32155ea867869a)

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**